src/app/clusters/device-energy-management-server/device-energy-management-server.cpp

```diff
@@ -0,0 +1,517 @@
+        return;
+    }
+
+    if (ESAStateEnum::kUserOptOut == mDelegate.GetESAState())
+    {
+        ChipLogError(Zcl, "DEM: %s - ESAState = kUserOptOut", __FUNCTION__);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    /* value SHALL be between the MinPauseDuration and MaxPauseDuration indicated in the
+       ActiveSlotNumber index in the Slots list in the Forecast.
+    */
+    uint16_t activeSlotNumber;
+    if (forecast.Value().activeSlotNumber.IsNull())
+    {
+        ChipLogError(Zcl, "DEM: %s - activeSlotNumber Is Null", __FUNCTION__);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    activeSlotNumber = forecast.Value().activeSlotNumber.Value();
+    if (activeSlotNumber >= forecast.Value().slots.size())
+    {
+        ChipLogError(Zcl, "DEM: %s - Bad activeSlotNumber %d , size()=%d.", __FUNCTION__, activeSlotNumber,
+                     static_cast<int>(forecast.Value().slots.size()));
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    if (!forecast.Value().slots[activeSlotNumber].slotIsPauseable)
+    {
+        ChipLogError(Zcl, "DEM: %s - activeSlotNumber %d is NOT pausible.", __FUNCTION__, activeSlotNumber);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    if ((duration < forecast.Value().slots[activeSlotNumber].minPauseDuration) &&
+        (duration > forecast.Value().slots[activeSlotNumber].maxPauseDuration))
+    {
+        ChipLogError(Zcl, "DEM: %s - out of range pause duration %d", __FUNCTION__, duration);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::ConstraintError);
+        return;
+    }
+
+    err = mDelegate.SetESAState(ESAStateEnum::kPaused);
+    if (CHIP_NO_ERROR != err)
+    {
+        ChipLogError(Zcl, "DEM: %s - SetESAState(paused) FAILURE", __FUNCTION__);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    /* Finally - we can call the delegate to ask for a pause */
+    status = mDelegate.PauseRequest(duration);
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+    if (status != Status::Success)
+    {
+        ChipLogError(Zcl, "DEM: %s - mDelegate.PauseRequest(%d) FAILURE", __FUNCTION__, duration);
+        return;
+    }
+
+    return;
+}
+
+void Instance::HandleResumeRequest(HandlerContext & ctx, const Commands::ResumeRequest::DecodableType & commandData)
+{
+    Status status = Status::Success;
+
+    DataModel::Nullable<Structs::ForecastStruct::Type> forecast = mDelegate.GetForecast();
+
+    if (ESAStateEnum::kPaused != mDelegate.GetESAState())
+    {
+        ChipLogError(Zcl, "DEM: %s - ESAState not Paused.", __FUNCTION__);
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::Failure);
+        return;
+    }
+
+    /* Call the delegate to resume */
```

Comment:
Comment not so useful ;-)
Similarly in other functions above ?

Reply:
Removed